group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1015,6 +1017,7 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
 
         // Get or create the consumer group.
         boolean createIfNotExists = memberEpoch == 0;
+        maybeUpgradeEmptyGroup(groupId, records, false);
```

Comment:
I think we can add check for migration config here which doesn't affect simple group upgrade.